webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -573,11 +573,21 @@ var userOverride = {
         var KM_ASK_USER_DETAILS = mckMessageService.checkArray(
             appOptions.askUserDetails
         );
-        var KM_PRELEAD_COLLECTION = mckMessageService.checkArray(
-            appOptions.preLeadCollection
-        )
-            ? appOptions.preLeadCollection
-            : [];
+        var KM_PRELEAD_COLLECTION;
+        if (appOptions.preLeadCollection) {
+            KM_PRELEAD_COLLECTION = mckMessageService.checkArray(
+                appOptions.preLeadCollection
+            );
```

Comment:
@Ojashwa when we write 
```js
let a = mckMessageService.checkArray(appOptions.preLeadCollection);
```
what is store in `a`?

Reply:
if the appOptions.preLeadCollection contain an array `a` will store that array and if not then it will throw an error `An array expected but got (whatever is the type of appOptions.preLeadCollection) `